src/v/config/configuration.h

```diff
@@ -342,6 +342,7 @@ struct configuration final : public config_store {
       partition_autobalancing_tick_interval_ms;
     property<size_t> partition_autobalancing_movement_batch_size_bytes;
     property<size_t> partition_autobalancing_concurrent_moves;
+    property<std::optional<size_t>> partition_autobalancing_min_size_threshold;
```

Comment:
Probably a good idea to include "full disk" in a name, but it is getting long...